docs/source/changelog.rst

```diff
@@ -4,23 +4,25 @@ Changelog
 ---------
 .. **Future Release**
     * Enhancements
+        * Add ``make_index`` to EntitySet init (:pr:`1004`)
```

Comment:
I think this could be clearer that it adds the ability to use `make_index` when initializing an EntitySet by passing in an `entities` dictionary